lib/ui/components/scenario-overview.js

```diff
@@ -47,6 +54,33 @@ export class ScenarioOverview extends Component {
     );
   }
 
+  onToggleRequest(requestId) {
+    this.props.toggleMock(this.props.id, requestId);
+  }
+
+  _renderMasterToggle() {
+    const toggleAllRequests = () => {
+      const nextState = this.state.masterToggle;
+      this.props.mockedRequests.forEach( request => {
+        if( (nextState && !request.active) || (!nextState && request.active ) ) {
+          this.onToggleRequest( request.id );
+        }
+      } );
+    };
+
+    return (
+      <div className="row">
+          <div className="col-xs-2 col-xs-offset-10">
+            <Toggle
+              disabled={ !this.props.active }
+              onToggle={ () => {
+                this.setState( { masterToggle : !this.state.masterToggle }, toggleAllRequests );
```

Comment:
There's no need to use the setState callback here, you can set the state in toggleAllRequests and then do the rest of the logic. there's no need to rely on when react actually updated the state